### DISCUSSION TITLE
Paging with updateQuery is deprecated?

### DISCUSSION THREAD

**Discussion by thomastvedt** [the-guild-org/apollo-angular#1651] [Q&A] [answered]
Hi, I've used apollo client earlier with React, and I'm trying out apollo-angular now. Currently implementing paging, and have some questions.

> If you are using Apollo-Client v3, please make sure to use apollo-angular@v2

I'm using these:
```
    "@apollo/client": "^3.3.12",
    "apollo-angular": "^2.4.0",
```

I'm  reading the [paging documentation](https://apollo-angular.com/docs/data/pagination) for V2. 
But the `updateQuery` callback is deprecated in Apollo client V3, and I get the following warning:

```
The updateQuery callback for fetchMore is deprecated, and will be removed
in the next major version of Apollo Client.

  Please convert updateQuery functions to field policies with appropriate
  read and merge functions, or use/adapt a helper function (such as
concatPagination, offsetLimitPagination, or relayStylePagination) from
@apollo/client/utilities.

  The field policy system handles pagination more effectively than a
hand-written updateQuery function, and you only need to define the policy
once, rather than every time you call fetchMore.
```

So, my question(s) are:
1. Is the documentation outdated? Is it possible to use field policies like they suggest?
2. What is the advantage of using the `apollo-angular` package? Can I just use the `@apollo/client` package? What do I get for free by using `apollo-angular`?
3. Is there a V3 version of apollo-angular planned? Does apollo-angular support everything in apollo/client v3.3? Or is version v2.4 === v3.3?
4. Do you have any examples using field policies? Or mentioned helper fundtions from `@apollo/client/utilities`?

**Comment by kamilkisiela** [comment, marked as answer]
1. The documentation is in fact outdated but you can check markdown files https://github.com/kamilkisiela/apollo-angular/blob/master/website/docs/data/pagination.md
2. You get support for RxJS, `SSR`, Angular `HttpClient` instead of Fetch etc
3. Apollo Angular v2 === Apollo Client v3 but there's a work in progress to release a new major version of AA (v3 === v3)
4. Yes, [cursor pagination](https://stackblitz.com/edit/apollo-angular-relay-cursor-pagination) and [offset pagination](https://stackblitz.com/edit/simple-apollo-angular-example-fetch-more?file=app%2Fgraphql.module.ts)
